data/domain/cards.tsx

```diff
@@ -148,6 +164,29 @@ export const updateCards = (data: Map<string, any>) => {
     const players = data.get("players") as Player[];
     const optLacc = data.get("OptLacc");
 
+    if (rift.bonuses.find(bonus => bonus.name == "Skill Mastery")?.active) {
+
+        const skillMastery = rift.bonuses.find(bonus => bonus.name == "Skill Mastery") as SkillMastery;
+
+        Object.entries(skillMastery.skillLevels).map(([skillAsString, skillLevel], index) => {
+            const skillName = SkillsIndex[parseInt(skillAsString)];
+            const skillIndex = SkillsIndex[skillName as keyof typeof SkillsIndex];
+
+            if (skillMastery.getSkillRank(skillIndex) >= 3 && SkillsforIDCardPassiveBonus[skillIndex].length > 0) {
+                cards.forEach(card => {
+                    if (SkillsforIDCardPassiveBonus[skillIndex].find(bonusID => bonusID == card.getBonusID())) {
+                        card.passive = true;
+                    }
+                })
+                players.flatMap(player => player.cardInfo?.cards ?? []).forEach(card => {
+                    if (SkillsforIDCardPassiveBonus[skillIndex].find(bonusID => bonusID == card.getBonusID())) {
+                        card.passive = true;
```

Comment:
Yeah the fact I duplicate cards is bad, so good job on catching it on both.

This can maybe be an alert in the future to let people know they are wasting a card slot by using a passive card 🤔